src/index/mod.rs

```diff
@@ -1 +1,68 @@
+use std::path::{Path, PathBuf};
+
+use log::info;
+use url::Url;
+
+use self::api::Api;
+use crate::error::Result;
+
 pub(crate) mod api;
+
+pub(crate) struct Index {
+    diff: crates_index_diff::Index,
+    path: PathBuf,
+    config: IndexConfig,
+}
+
+#[derive(serde::Deserialize, Clone)]
+#[serde(rename_all = "kebab-case")]
+struct IndexConfig {
```

Comment:
This is clever! `serde` is so useful :D

Reply:
Serde is great, I totally recommend reading over all their [attributes](https://serde.rs/attributes.html). Every time I look at it I discover something new